src/csharp/Microsoft.Spark/Pyrolite/UnpickleStack.cs

```diff
@@ -16,7 +16,7 @@ namespace Razorvine.Pickle
 [SuppressMessage("ReSharper", "InconsistentNaming")]
 internal class UnpickleStack {
 	public readonly object MARKER;
-    private const int DefaultCapacity = 4;
+    private const int DefaultCapacity = 128;
```

Comment:
this small change was very important: the unpickler adds every deserialized object to this collection.

if we deserialize 100 doubles, it would allocate an array of 4 objects, double it to 8 (and copy all elements), 16, 32, 64, 128 etc..

128 is better because we hardly ever want to deserialize and send only a few objects from one process to another